pkg/manager/impl/validation/validation_test.go

```diff
@@ -334,3 +359,88 @@ func TestValidateOutputData(t *testing.T) {
 		assert.Equal(t, codes.ResourceExhausted, err.(errors.FlyteAdminError).Code())
 	})
 }
+
+func TestValidateDatetime(t *testing.T) {
+	t.Run("no datetime", func(t *testing.T) {
+		assert.EqualError(t, ValidateDatetime(nil), "Found invalid nil datetime")
+	})
+	t.Run("datetime with valid format and value", func(t *testing.T) {
+		assert.NoError(t, ValidateDatetime(&core.Literal{
+			Value: &core.Literal_Scalar{
+				Scalar: &core.Scalar{
+					Value: &core.Scalar_Primitive{
+						Primitive: &core.Primitive{
+							Value: &core.Primitive_Datetime{Datetime: timestamppb.Now()},
+						},
+					},
+				},
+			},
+		}))
+	})
+	t.Run("datetime with value below min", func(t *testing.T) {
+		timestamp := timestamppb.Timestamp{Seconds: -62135596801, Nanos: 999999999} // = 0000-12-31T23:59:59.999999999Z
+		expectedErrStr := "proto: timestamp (seconds:-62135596801 nanos:999999999) before 0001-01-01"
+		expectedErrStrWithValidFormat := strings.Replace(expectedErrStr, " ", "\\u00a0", 0)
+		expectedErr := errors.NewFlyteAdminErrorf(codes.InvalidArgument, "TestValidateDatetime/datetime_with_value_below_min")
+
+		assert.EqualError(t, ValidateDatetime(&core.Literal{
+			Value: &core.Literal_Scalar{
+				Scalar: &core.Scalar{
+					Value: &core.Scalar_Primitive{
+						Primitive: &core.Primitive{
+							Value: &core.Primitive_Datetime{Datetime: &timestamp},
+						},
+					},
+				},
+			},
+		}), expectedErrStrWithValidFormat, expectedErr)
+	})
+	t.Run("datetime with min valid value", func(t *testing.T) {
+		timestamp := timestamppb.Timestamp{Seconds: -62135596800, Nanos: 0} // = 0001-01-01T00:00:00Z
+
+		assert.NoError(t, ValidateDatetime(&core.Literal{
+			Value: &core.Literal_Scalar{
+				Scalar: &core.Scalar{
+					Value: &core.Scalar_Primitive{
+						Primitive: &core.Primitive{
+							Value: &core.Primitive_Datetime{Datetime: &timestamp},
+						},
+					},
+				},
+			},
+		}))
+	})
+	t.Run("datetime with max valid value", func(t *testing.T) {
+		timestamp := timestamppb.Timestamp{Seconds: 253402300799, Nanos: 999999999} // = 9999-12-31T23:59:59.999999999Z
+
+		assert.NoError(t, ValidateDatetime(&core.Literal{
+			Value: &core.Literal_Scalar{
+				Scalar: &core.Scalar{
+					Value: &core.Scalar_Primitive{
+						Primitive: &core.Primitive{
+							Value: &core.Primitive_Datetime{Datetime: &timestamp},
+						},
+					},
+				},
+			},
+		}))
+	})
+	t.Run("datetime with value above max", func(t *testing.T) {
+		timestamp := timestamppb.Timestamp{Seconds: 253402300800, Nanos: 0} // = 0000-12-31T23:59:59.999999999Z
```

Comment:
let's also add a test for the java minimum (which should not be supported)

```
MIN
The minimum supported Instant, '-1000000000-01-01T00:00Z'.

Reply:
Done!